### ISSUE TITLE
FlareSolverr isnt work correctly

### ISSUE THREAD

**Issue by Infel1** [HDoujinDownloader/HDoujinDownloader#366]
FlareSolverr is not working properly. At least I think so. In the video you can see that at the bottom the word "waiting for Cloudflare" blinks and that causes the download to slow down too much. Also in the CMD video I manually ran FlareSolverr and then started downloading a Manhua at "https://tresdaos.com/" and it gave me the following error messages:

https://github.com/HDoujinDownloader/HDoujinDownloader/assets/36248950/87f5e862-022a-41f7-b478-148bd346205e

https://github.com/HDoujinDownloader/HDoujinDownloader/assets/36248950/043f76ba-390b-4e52-a41f-52155e6e3b83

**Comment by HDoujinDownloader** [comment]
I'm not totally sure what's going with this website because I can't access it right now (it appears to be down). Cloudflare can be triggered for a number of reasons, some of which are easier to fix than others, so I'll need to investigate when it's back online. Let me know when it's back up.

FlareSolverr in general is kind of iffy right now, though, but the FlareSolverr team is aware of it: FlareSolverr/FlareSolverr#1036

There's a WIP fix (FlareSolverr/FlareSolverr#1163), but I don't think there's an ETA for when that will be available yet. 

Depending on what's triggering Cloudflare in this scenario, there is also the option to bypass it manually by copying the clearance cookies from your web browser: https://github.com/HDoujinDownloader/HDoujinDownloader/wiki/Cloudflare#bypassing-cloudflare-manually

**Comment by Infel1** [comment]
@HDoujinDownloader Man, the website really went down. When it's back online, I'll let you know. Hdoujin also has some other bug that I will inform you about later. I hope FlareSolverr fixes that problem because it's honestly annoying, I won't judge them because they do the best they can and the program is relatively new. Thanks for always paying attention Squidy.

**Comment by Infel1** [comment]
@HDoujinDownloader Man, the website really went down. When it's back online, I'll let you know. Hdoujin also has some other bug that I will inform you about later. I hope FlareSolverr fixes that problem because it's honestly annoying, I won't judge them because they do the best they can and the program is relatively new. Thanks for always paying attention Squidy.

**Comment by HDoujinDownloader** [comment]
@Infel1 Yeah, it's a cat-and-mouse game between Cloudflare and FlareSolverr, the battle never ends.

It looks like the website is back up, and I found some chapters that have the problem you're describing. From my tests it seems to happen for images that are hosted on "i.yapx.ru". I can't get the images to load in my browser either if I open the chapters there, though (maybe it's a temporary ban after failing to download them with HDoujin Downloader?)-- Let me know if your experience is different with this.

I'm thinking FlareSolverr shouldn't be fired up for this scenario though as it's unlikely to resolve the block. I'll see if I can add a check for this to prevent it from slowing down the downloads so much when the images fail.

**Comment by Infel1** [comment]
Did you fix the bug or what happened? Now the Cloudflare bug does not appear when I go to download in tresdaos. @HDoujinDownloader

**Comment by Infel1** [comment]
@HDoujinDownloader  Did you fix the Bug or what happened? Now the Cloudflare Bug does not appear when I go to download in tresdaos.

 My bad, it seems like you said in some images or chapters the Cloudflare Bug appears for some reason. Just now I was downloading two manhuas and out of nowhere the Bug appeared

**Comment by HDoujinDownloader** [comment]
If you try to look at the images for affected chapters in your web browser, do you find they fail to load there as well? I think it might be a problem with their image host, but I want to make sure.

**Comment by Infel1** [comment]
@HDoujinDownloader I realized that the problems were with the server, there were images that did not work or that they were hosted in Russia. Perhaps that is why Hdoujin had problems downloading them because he did not know how to detect those chapters hosted in Russia and because some had not been uploaded or for some reason were deleted.
